internal/ingress/controller/parser/parser.go

```diff
@@ -942,7 +943,18 @@ func overrideRouteByKongIngress(route *Route,
 
 	r := kongIngress.Route
 	if len(r.Methods) != 0 {
-		route.Methods = cloneStringPointerSlice(r.Methods...)
+		invalid := false
+		for _, method := range r.Methods {
+			if !validMethods.MatchString(*method) {
```

Comment:
Let's do two things to make this user friendly:
- upper case the string always and allow for lower-cased method names
- trim space, prefix and suffix both

Add a test-case for both as well.